apisix/discovery/kubernetes/init.lua

```diff
@@ -324,23 +333,31 @@ local function get_apiserver(conf)
             return nil, err
         end
     elseif conf.client.token_file and conf.client.token_file ~= "" then
-        local file
-        file, err = read_env(conf.client.token_file)
-        if err then
-            return nil, err
-        end
+        setmetatable(apiserver, {
+            __index = function(_, key)
+                if key ~= "token" then
```

Comment:
can we use same way like https://github.com/apache/apisix/blob/8179784fe68f67fbf79e0110ba8f6f3a5ad56c63/apisix/plugins/body-transformer.lua#L176